src/components/elements/TitleCard.tsx

```diff
@@ -60,6 +62,7 @@ const TitleCard: React.FC<Props> = ({
         }}
       >
         <Typography
+          component={headerComponent || 'span'}
```

Comment:
Defaulting to `span` might not be right. Can we just leave it empty if `headerComponent` was not passed?
It will set the component based on the default `variantMapping` https://mui.com/material-ui/api/typography/ so maybe we should leave that behavior as-is until we address the header tag issue across the board